tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,760 @@
+  'File' => absolute path of the file with the given ID.
+
+=cut
+sub getIntermediaryFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # This should get all files in table files_intermediary that are tied
+    # indirectly to the tarchive with ID $tarchiveId
+    my $query = 'SELECT fi.IntermedID, f.FileID, f.File FROM files_intermediary fi '
+        .       'JOIN files f ON (fi.Output_FileID=f.FileID) '
+        .       'WHERE f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)';
+
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+        $_->{'File'} = "$dataDirBasePath/$_->{'File'}" unless $_->{'File'} =~ /^\//;
+    } 
+    
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getParameterFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Gets the absolute paths of all the files associated to an archive 
+that are listed in table C<parameter_file> and have a parameter
+type set to C<check_pic_filename>.
+
+INPUTS:
+
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting 'dataDirBasePath'.
+
+RETURNS: 
+
+  an array of hash references. Each hash has two keys: 'FileID' => FileID of a file 
+  in table parameter_file and 'Value' => absolute path of the file with the given ID.
+
+=cut
+sub getParameterFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath) = @_;
+    
+    # Get all files in parameter_file with parameter_type set to
+    # 'check_pic_filename' that are tied (indirectly) to the tarchive
+    # with ID $tarchiveId
+    (my $query =<<QUERY) =~ s/\s+/ /g;
+        SELECT FileID, Value FROM parameter_file pf
+        JOIN files f USING (FileID)
+        JOIN parameter_type AS pt
+        USING (ParameterTypeID)
+        WHERE pt.Name='check_pic_filename'
+        AND (
+                f.TarchiveSource = ? 
+             OR f.SourceFileID IN (SELECT FileID FROM files WHERE TarchiveSource = ?)
+        )
+QUERY
+
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID, $tarchiveID);
+    
+    # Make sure all paths are absolute paths
+    foreach(@$filesRef) {
+        if($_->{'Value'} !~ /^\//) {
+            $_->{'Value'} = sprintf("%s/%s/%s", $dataDirBasePath, PIC_SUBDIR, $_->{'Value'});
+        }
+    }
+    
+    return $filesRef;
+}
+
+=pod
+
+=head3 getMriProtocolViolatedScansFilesRef($dbh, $tarchiveID, $dataDirBasePath)
+
+Get the absolute paths of all the files associated to an archive that are listed in 
+table C<mri_protocol_violated_scans>.
+
+INPUTS:
+
+  - $dbhr  : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting 'dataDirBasePath'.
+
+RETURNS: 
+
+ an array of hash references. Each hash has one key: 'minc_location' => location (absolute path)
+ of a MINC file found in table mri_protocol_violated_scans.
```

Comment:
mri_protocol_violated_scans should be in C<>.

Reply:
Ok